docs/contributing/dev_env.rst

```diff
@@ -64,6 +64,15 @@ their latest versions (``pip setuptools wheel``).
     $ python3 -m pip install -U pip setuptools wheel
     $ python3 -m pip install --prefix=~/.local -e .[dev]
 
+
+In case you're using zsh shell, you may encounter ``zsh: no matches found: [dev]``. 
+This occurs as zsh interprets square brackets differently. This issue can be 
+resolved by enclosing [dev] within quotation marks
+
+.. code-block:: console
+
+    $ python3 -m pip install --prefix=~/.local -e .'[dev]'
```

Comment:
Looks great !! Have you verified this in a zsh env. Just for sanity, please copy this as it is and make sure if it works :)

Reply:
Thanks! Yeah, I made sure it works on my zsh env.